### PR TITLE
Use the assigned schema validation results

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary

If you use any transformers in yup, we don't use the return value. So we can't use things like default(), or trim().

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/fileStores/keyStore.ts [hunk 1]
```diff
@@ -48,10 +48,15 @@ export class KeyStore<TSchema extends Record<string, unknown>> {
...
     // Validate file store if we have a schema
     if (this.schema) {
-      const { error } = await YupUtils.tryValidate(this.schema, data)
+      const { error, result } = await YupUtils.tryValidate(this.schema, data)
+
       if (error) {
         throw new Error(error.message)
       }
+
+      if (result) {
```
Comment (danield9tqh) [inline ironfish/src/fileStores/keyStore.ts:57]: `result` and `error` are mutually exclusive on this return type right? Do we need to do this check here or can we just assign?

Comment (NullSoldier) [reply ironfish/src/fileStores/keyStore.ts:57]: Good find, I didn't see that it was typed like that.

File: ironfish/src/fileStores/keyStore.ts [hunk 2]
```diff
@@ -48,10 +48,15 @@ export class KeyStore<TSchema extends Record<string, unknown>> {
...
+        Object.assign(data, result)
```
Comment (danield9tqh) [inline ironfish/src/fileStores/keyStore.ts:58]: Also I think we might need to check if `data === null` before this. If the file doesn't exist.

Comment (NullSoldier) [reply ironfish/src/fileStores/keyStore.ts:58]: added check

**Review by danield9tqh — APPROVED** [review]

**Pull request merged**